doc/install.rst

```diff
@@ -1,58 +1,655 @@
+    more efficient code.
 
-  * Sphinx
+    The main computational loops in Arbor are based on
+    `NMODL <https://www.neuron.yale.edu/neuron/static/docs/help/neuron/nmodl/nmodl.html>`_,
+    from which Arbor generates vectorized code, which can be compiled very efficiently by GCC
+    and Clang.
+    This allows us focus on GCC and Clang for faster compilation times, fewer
+    compiler bugs to work around, and support for recent C++ standards.
+
+.. Note::
+    The IBM xlc compiler versions 13.1.4 and 13.1.6 have been tested for compiling on
+    IBM power 8. Arbor contains some patches to work around xlc compiler bugs,
+    however we do not recommend using xlc because GCC produces faster code,
+    with lower comilation times.
+
+Optional Requirements
+---------------------
+
+GPU Support
+~~~~~~~~~~~
+
+Arbor has full support for NVIDIA GPUs, for which the NVIDIA CUDA toolkit version 8 is required.
+
+Distributed
+~~~~~~~~~~~
+
+Arbor uses MPI to run on HPC cluster systems.
+Arbor has been tested on MVAPICH2, OpenMPI, Cray MPI, and IBM MPI.
+More information on building with MPI is in the `HPC cluster section <cluster_>`_.
+
+Documentation
+~~~~~~~~~~~~~~
+
+To build a local copy of the html documentation that you are reading now, you will need to
+install `Sphinx <http://www.sphinx-doc.org/en/master/>`_.
 
 .. _downloading:
 
-Downloading
-======================================
+Getting the Code
+================
+
+The easiest way to acquire the latest version of Arbor is to check the code out from
+the `Github repository <https://github.com/eth-cscs/arbor>`_:
+
+.. code-block:: bash
+
+    git clone https://github.com/eth-cscs/arbor.git --recursive
 
-The easiest way to acquire the latest version of Arbor is to check the code out from our GitHub repository:
+We recommend using a recursive checkout, because Arbor uses git submodules for some
+of its library dependencies.
+The CMake configuration attempts to detect if a required submodule is available, and
+will print a helpful warning
+or error message if not, but it is up to the user to ensure that all required
+submodules are downloaded.
+
+The git submodules can be updated, or initialized in a project that didn't use a
+recursive checkout:
 
 .. code-block:: bash
 
-    git clone https://github.com/eth-cscs/arbor.git
+    git submodule update --init --recursive
 
-You can also point your browser to our `Github page <https://github.com/eth-cscs/arbor>`_ and download a zip file.
+You can also point your browser to Arbor's
+`Github page <https://github.com/eth-cscs/arbor>`_ and download a zip file.
+If you use the zip file, then don't forget to run git submodule update manually.
 
-.. _install_desktop:
+.. _building:
 
-Basic Installation
-======================================
+Building Arbor
+==============
 
 Before building an optimzed version for your target system, it is a good idea to build a debug version:
```

Comment:
It's not clear why this is a good idea.

I'd propose stating that:
* Arbor can be built in debug or release mode.
* Debug mode is useful for troubleshooting (link to section).
* Release mode should be used for installation, benchmarking etc.

Reply:
Things are more likely to break and tests are more likely to fail when compiling with optimization on (see our experiences with Power 8 and Arm). I wanted to have a "most likely to work" set of steps that use the least number of flags presented up front.

I agree that this sentence is a misleading, because obviously if you want to use/install Arbor a debug build is a terrible idea.

I have reworded things, and used release mode for that example workflow.